### DISCUSSION TITLE
How to manage term ambiguity, term mismatch or term synonym

### DISCUSSION THREAD

**Discussion by TheMrguiller** [memgraph/memgraph#2470] [General]
Hi,

I have a query about what tools Memgraph offers to solve or address term ambiguity, term mismatch, or term synonymy. We are dealing with these problematics as the database knowledge and expert knowledge may diverge in the wording used to address several entities like organizations or names. Especially, it has become a problem when dealing with LLM-powered applications, making us think about generating a glossary of terms. We wonder if there is any other alternative that Memgraph can offer for these cases, such as term similarity or partial match, among others.

Thanks for the help

**Comment by matea16** [comment]
Hi @TheMrguiller, Memgraph currently doesn’t offer a direct solution for handling term ambiguity, mismatch, or synonymy, but this is a known challenge in LLM-based applications. One approach you might consider is using an external library that can identify term similarity or partial matches. Memgraph is also working on adding vector search capabilities, which could help in identifying similar nodes with embeddings, which might help. Have you experimented with any external libraries to manage term matching so far?
